main.c

```diff
@@ -330,39 +326,40 @@ static int socket_bindlisten(const char *socket_path,
   memset(&addr, 0, sizeof(addr));
   unlink(socket_path); /* avoid EADDRINUSE */
   if ((fd = socket(PF_LOCAL, SOCK_STREAM, 0)) < 0) {
-    perror("socket");
+    ERRORN("socket");
```

Comment:
ERRORN was a quick way to convert perror calls to standard logs, but I think in most case it will be better to use ERRORF and add more context about the error. For example here adding the socket path will help to debug issues.

Reply:
Can be improved later.